shapely/tests/geometry/test_geometry_base.py

```diff
@@ -157,3 +158,104 @@ def test_constructive_properties(op):
     result = getattr(geom, op)
     expected = getattr(shapely, op)(geom)
     assert result == expected
+
+
+@pytest.mark.parametrize(
+    "op",
+    [
+        "crosses",
+        "contains",
+        # "contains_properly",
+        "covered_by",
+        "covers",
+        "disjoint",
+        "equals",
+        "intersects",
+        "overlaps",
+        "touches",
+        "within",
+    ],
+)
+def test_array_argument_binary_predicates(op):
+    polygon = Polygon([(0, 0), (0, 1), (1, 1), (1, 0), (0, 0)])
+    points = shapely.points([(0, 0), (0.5, 0.5), (1, 1)])
+
+    result = getattr(polygon, op)(points)
+    assert isinstance(result, np.ndarray)
+    expected = np.array([getattr(polygon, op)(p) for p in points], dtype=bool)
+    np.testing.assert_array_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "op, kwargs",
+    [
+        ("dwithin", dict(distance=0.5)),
+        ("equals_exact", dict(tolerance=0.01)),
+        ("relate_pattern", dict(pattern="T*F**F***")),
+    ],
+)
+def test_array_argument_binary_predicates2(op, kwargs):
+    polygon = Polygon([(0, 0), (0, 1), (1, 1), (1, 0), (0, 0)])
+    points = shapely.points([(0, 0), (0.5, 0.5), (1, 1)])
+
+    result = getattr(polygon, op)(points, **kwargs)
+    assert isinstance(result, np.ndarray)
+    expected = np.array([getattr(polygon, op)(p, **kwargs) for p in points], dtype=bool)
+    np.testing.assert_array_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "op",
+    [
+        "difference",
+        "intersection",
+        "symmetric_difference",
+        "union",
+    ],
+)
+def test_array_argument_binary_geo(op):
+    box = Polygon([(0, 0), (0, 1), (1, 1), (1, 0), (0, 0)])
+    polygons = shapely.buffer(shapely.points([(0, 0), (0.5, 0.5), (1, 1)]), 0.5)
+
+    result = getattr(box, op)(polygons)
```

Comment:
I added tests for the overlay set ops as wel, but just to note that those actually already behaved this way (didn't change the implementation)